text/0039-service-overhaul.md

```diff
@@ -0,0 +1,196 @@
+### Issue
+#### `rtmp-services`
+This plugin provide services with only one protocol, so it need to "register" service like YouTube multiple times (HLS & RTMPS).
+
+And also OBS is too heavily dependent of it.
+
+#### Service integrations
+Those integrations are added and use `rtmp-services` in a hacky way and not as plugins.
+
+### Idea
+#### `rtmp-services`
+This plugin will be put in a state of deprecation with a legacy behavior to not break immediately every setup. And be replaced and finally removed after one major release cycle.
+
+Since OBS is heavily dependent of this plugin, many changes will be needed to remove this dependency.
+
+#### Service integrations
+
+Twitch, YouTube and Restream have their own integration in OBS with OAuth. But those need to be isolated as front-end plugin(s).
+
+With an updated front-end API for those and future third-party plugins.
+
+#### `custom-service`
+This plugin is meant to provide a replacement for `rtmp_custom` type.
+
+If OBS need to be heavily dependent to one plugin, it shall be this one.
+
+The protocol will be detected (or maybe forced set by the user).
+Maybe properties will show depending of the protocol.
+
+#### `obs-services`
+This plugin is meant to provide a replacements for `rtmp_common` type for services who doesn't need custom behavior.
+
+This plugin will need a brand new services.json with new format, to register each streaming service with their own id. No more things like `rtmp-common` id.
+ 
+Those services will be able to provide multiple protocols so no more "Service - HLS" and "Service - FTL".
+
+If a certain protocol is not available, the plugin will not register the service if the service doesn't use another protocol.
+If it does, the missing protocol will not be shown. [Thanks to RFC 38](0038-protocol-api.md).
+
+The notion of maximum and recommended shall be distinguished.
+Maximum, supported resolutions, recommended settings will be ignorable with their respective checkboxes.
+
+Those services should have no specific behavior like ingest management.
+
+But can list with which codec they are compatible if needed. [Thanks to RFC 38](0038-protocol-api.md)
+
+In the end, adding streaming service will only be a JSON addition for this plugin.
+Only improvements will be accepted in the code of this plugin.
+
+#### `obs-services2`
+This plugin is meant to provide a replacements for `rtmp_common` type for services who need a custom behavior.
+
+This plugin is meant to have implemented and register services which need a specific behavior like ingest with a unique id for each one.
+
+No JSON file will be used to create those services, except maybe for servers list.
+
+#### Note about the plugins
+- Like this if the service needs a specific behavior, you can 'transfer' it from `obs-services` to `obs-services2` and keep the same id. The change will be seamless for the user.
+- Services with integrations may have two services registered. The basic one and its counterpart with OAuth.
```

Comment:
It might be cleaner to have a way to support both from the same service. This way users get confused less and there is less clutter.

Reply:
>I wanted to make them separated to be able to extract those OAuth plugins from the PPA packages to provide them as a standalone package for Linux distro with unofficial packages.
>
>But this is not part of RFC. And can be done an other way.

We can make plugins providing both way in one service with a switch button in their properties views. And not providing the client ids will build them without the integration part.